### ISSUE TITLE
Timeseries metrics seems to have conflicting data

### ISSUE THREAD

**Issue by ryscheng** [opensource-observer/oso#2377]
### What is it?

From Elliot

When running a query I get different amounts returned for the same metricId, ProjectId and sampleDate.
Not sure if my expectations are wrong or if this is just because the data isn't quite ready.  
JavaScript

```
1: {
        amount: 1,
        metricId: 'eziTctwFeZugH0ocp4pjj9cXQFTRVCm6KgJ+ozZCmmQ=', 
        projectId: 'cLZx_e_EZ-UuWqfOXvqTyUZfecyfWGmbQnA2lRMKrR8=', 
        sampleDate: '2024-04-23', 
        unit: null
    },
2: {
        amount: 115, 
        metricId: 'eziTctwFeZugH0ocp4pjj9cXQFTRVCm6KgJ+ozZCmmQ=', 
        projectId: 'cLZx_e_EZ-UuWqfOXvqTyUZfecyfWGmbQnA2lRMKrR8=', 
        sampleDate: '2024-04-23', 
        unit: null
    }
```

Found with this query: 
```
oso_timeseriesMetricsByProjectV0(where: $where) {
      amount
      metricId
      projectId
      sampleDate
      unit
    }
```
with a where clause that looks like this:
```
where: {
      projectId: {
        _eq: projectId,
      },
      metricId: {
        _in: metricIds,
      },
      sampleDate: {
        _gt: startDate,
        _lte: endDate,
      },
    },
```
I would expect that there would only be one entry per unique metricId, projectId and sampleDate combinations.

Thanks for helping me understand why this is the case.

**Comment by ravenac95** [comment]
Technically we fix the issue in #2400 but this will take a little while to get reflected on the api.

**Comment by ravenac95** [comment]
Thanks @escottalexander for this issue!
